src/file.ts

```diff
@@ -1992,7 +1990,12 @@ class File extends ServiceObject<File> {
         .on('error', callback)
         .on('finish', callback);
     } else {
-      fileStream.on('error', callback).pipe(concat(callback.bind(null, null)));
+      getStream
+        .buffer(fileStream)
+        .then(
```

Comment:
This might be easier to read if it was something like:

```js
getStream
  .buffer(fileStream)
  .then(contents => callback(null, contents))
  .catch(callback)
```

I know you copied our existing `callback.bind(null, null)` (thank you for the attention to detail), but I think arrow functions are new since we wrote this, so we can "upgrade" now, if you're willing :)

Reply:
Not really easier. Refinement will be invalidated and callback will become nullable again.

Reply:
True, I think TS makes both options much less readable :) I think "bind" was a well-known, useful solution in the old days, but I don't know how many generations of developers are left who will see that and know what's really happening or why. And with arrow functions being an option, I don't think we have a good explanation anymore. Thanks for making the change.

Reply:
I used optional chaining to simplify this. Looks like everything still works.